src/plugins/orangehrmPerformancePlugin/Api/EmployeeTrackerAPI.php

```diff
@@ -0,0 +1,203 @@
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Performance\Api;
+
+use OrangeHRM\Core\Api\CommonParams;
+use OrangeHRM\Core\Api\V2\CollectionEndpoint;
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\EndpointCollectionResult;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\ParameterBag;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Core\Traits\Auth\AuthUserTrait;
+use OrangeHRM\Performance\Api\Model\EmployeeTrackerModel;
+use OrangeHRM\Performance\Dto\EmployeeTrackerSearchFilterParams;
+use OrangeHRM\Performance\Service\EmployeeTrackerService;
+
+class EmployeeTrackerAPI extends Endpoint implements CollectionEndpoint
+{
+    use AuthUserTrait;
+
+    public const FILTER_INCLUDE_EMPLOYEES = 'includeEmployees';
+    public const FILTER_NAME_OR_ID = 'nameOrId';
+
+    public const PARAM_RULE_FILTER_NAME_OR_ID_MAX_LENGTH = 100;
+
+    private const ADMIN_USER_ROLE_ID = 1;
+
+    private ?EmployeeTrackerService $employeeTrackerService = null;
+
+    /**
+     * @return EmployeeTrackerService
+     */
+    public function getEmployeeTrackerService(): EmployeeTrackerService
+    {
+        if (!$this->employeeTrackerService instanceof EmployeeTrackerService) {
+            $this->employeeTrackerService = new EmployeeTrackerService();
+        }
+        return $this->employeeTrackerService;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getAll(): EndpointResult
+    {
+        $employeeTrackerSearchFilterParams = $this->getEmployeeTrackerSearchFilterParams();
+        $this->setSortingAndPaginationParams($employeeTrackerSearchFilterParams);
+
+        $isAdmin = $this->getAuthUser()->getUserRoleId() === self::ADMIN_USER_ROLE_ID;
+        if (!$isAdmin) {
+            $empNumber = $this->getAuthUser()->getEmpNumber();
+            $employeeTrackerList = $this->getEmployeeTrackerService()
+                ->getEmployeeTrackerDao()
+                ->getEmployeeTrackerListForESS($employeeTrackerSearchFilterParams, $empNumber);
+            $employeeTrackerCount = $this->getEmployeeTrackerService()
+                ->getEmployeeTrackerDao()
+                ->getEmployeeTrackerCountForESS($employeeTrackerSearchFilterParams, $empNumber);
+        } else {
+            $employeeTrackerList = $this->getEmployeeTrackerService()
+                ->getEmployeeTrackerDao()
+                ->getEmployeeTrackerListForAdmin($employeeTrackerSearchFilterParams);
+            $employeeTrackerCount = $this->getEmployeeTrackerService()
+                ->getEmployeeTrackerDao()
+                ->getEmployeeTrackerCountForAdmin($employeeTrackerSearchFilterParams);
+        }
```

Comment:
without hard coding user role based if conditions, hand over it to `$this->getUserRoleManager()->getAccessibleEntityIds(Employee::class)`. Based on resource you need to change the entity class name. 
e.g. `$this->getUserRoleManager()->getAccessibleEntityIds(EmployeeTraker::class)`

ref: https://github.com/orangehrm/orangehrm/blob/main/src/plugins/orangehrmCorePlugin/Authorization/UserRole/AdminUserRole.php#L58

#### How to handle handle filter;
https://github.com/orangehrm/orangehrm/blob/main/src/plugins/orangehrmPimPlugin/Api/EmployeeAPI.php#L163-L168

Reply:
@RajithaKumara could you check my current approach?